### ISSUE TITLE
Cannot make sense of Memory usage at the top and net memory consumption

### ISSUE THREAD

**Issue by Akashdesarda** [plasma-umass/scalene#221]
**Describe the bug**
I am profiling a simple pandas code where it reads an excel file (about 2400+) and converts them into csv file. Memory usage at the top is showing as max:  93.47GB, whereas my machine has 16 gb of memory. I am not saying that this might be a bug (or it can be?), I just want to make sense of this report. 

                                        Memory usage: ██████████ (max:  93.47GB, growth rate:   6%)
                                             pre_ppa.py: % of time = 100.00% out of 8599.11s.
           ╷       ╷       ╷       ╷        ╷       ╷               ╷       ╷
      Line │Time   │–––––– │–––––– │Memory  │–––––– │–––––––––––    │Copy   │
           │Python │native │system │Python  │net    │timeline/%     │(MB/s) │pre_ppa.py
    ╺━━━━━━┿━━━━━━━┿━━━━━━━┿━━━━━━━┿━━━━━━━━┿━━━━━━━┿━━━━━━━━━━━━━━━┿━━━━━━━┿━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸

       105 │   65% │    9% │       │  61%   │71.91G │▅▅▅▅▅▅  62%    │    20 │    data = pd.read_excel(file, sheet_name="Input_PPA")

```
╶──────┼───────┼───────┼───────┼────────┼───────┼───────────────┼───────┼─────────────────────────────────────────────────────────╴
           │       │       │       │        │       │               │       │function summary for pre_ppa.py
        11 │       │       │       │        │       │               │       │set_logging
        78 │       │       │       │        │    7M │█████          │       │ppa_data
        93 │   65% │    9% │       │  61%   │71.97G │████████▂  62% │    20 │ingestion_test
       135 │    3% │       │  17%  │   4%   │  985M │██████▁▁▁   4% │     1 │upload_data
           ╵       ╵       ╵       ╵        ╵       ╵               ╵       ╵
    Top net memory consumption, by line:
    (1)   105: 73637 MB
    (2)   169: 18009 MB
    (3)   181:  2523 MB
    (4)   153:   740 MB
    (5)   200:   414 MB
```

**To Reproduce**
If you want I can email you the HTML report 

**Expected behavior**
Showing accurate memory consumption which certenly should be between 1-16 gb (as the code did successfully ran on my machine)

**Desktop (please complete the following information):**
 - OS: RHEL 8
 - Version 1.3.6

**Additional context**

**Comment by emeryberger** [comment]
Can you provide the source code? (You can mail me if you'd rather not post it publicly.)

**Comment by Akashdesarda** [comment]
> Can you provide the source code? (You can mail me if you'd rather not post it publicly.)

I can send you the HTML report as it will contain the complete source code and line by line analysis. Is this your email- emery.berger@gmail.com? I have seen that you have provided it in some other issue thread.

**Comment by emeryberger** [comment]
Yes - but please also send the raw source code if you can - thanks!

**Comment by dralley** [comment]
@emeryberger I'm running into a similar issue (on Fedora Linux, if that's relevant), and I can provide source code if that would be helpful.
